benches/profile/process_test_utils.py

```
@@ -0,0 +1,54 @@
+import pathlib
+import threading
+import time
+
+
+def read_with_timeout(stream, operation, timeout_seconds=5):
+    result = queue.Queue(maxsize=1)
+
+    def read():
+        try:
+            result.put((True, operation(stream)))
+        except BaseException as error:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `BaseException`

(BLE001)